File: queue.go
package main

import (
	"fmt"
	"math/rand"
	"strconv"
	"strings"
)

// Генерация номера талона (как в оригинале)
func generateTicket() string {
	return "T" + strconv.Itoa(rand.Int()%1000)
}

func main() {
	var windows int
	fmt.Print(">>> Введите кол-во окон\n<<< ")
	fmt.Scan(&windows)

	// Посетители: слайс пар (время, талон)
	visitors := make([][2]interface{}, 0)
	windowTimes := make([]int, windows)
	windowTickets := make([][]string, windows)

	var cmd string
	for {
		fmt.Print("<<< ")
		fmt.Scan(&cmd)

		switch cmd {
		case "ENQUEUE":
			var time int
			fmt.Scan(&time)
			ticket := generateTicket()
			visitors = append(visitors, [2]interface{}{time, ticket})
			fmt.Printf(">>> %s\n", ticket)

		case "DISTRIBUTE":
			// Распределение посетителей по окнам
			for _, visitor := range visitors {
				// Находим окно с минимальным временем
				minIndex := 0
				for i := 1; i < windows; i++ {
					if windowTimes[i] < windowTimes[minIndex] {
						minIndex = i
					}
				}

				// Добавляем посетителя в окно
				time := visitor[0].(int)
				ticket := visitor[1].(string)
				windowTimes[minIndex] += time
				windowTickets[minIndex] = append(windowTickets[minIndex], ticket)
			}

			// Вывод результатов
			for i := 0; i < windows; i++ {
				if len(windowTickets[i]) > 0 {
					fmt.Printf(">>> Окно %d (%d минут): %s\n",
						i+1,
						windowTimes[i],
						strings.Join(windowTickets[i], ", "))
				}
			}
			return

		default:
			fmt.Println(">>> Неизвестная команда")
		}
	}
}

File: queue.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

// Генерация номера талона
string generateTicket() {
    return "T" + to_string(rand() % 1000);
}

// Функция распределения очереди по окнам
void distribute(const vector<pair<int, string>>& visitors, vector<int>& windowTimes, vector<vector<string>>& windowTickets) {
    // Распределяем посетителей по окнам
    for (const auto& visitor : visitors) {
        // Находим окно с минимальным временем
        int minWindow = min_element(windowTimes.begin(), windowTimes.end()) - windowTimes.begin();
        windowTimes[minWindow] += visitor.first;
        windowTickets[minWindow].push_back(visitor.second);
    }
}

int main() {
    cout << ">>> Введите кол-во окон\n<<< ";
    int windows;
    cin >> windows;
    
    vector<pair<int, string>> visitors;
    vector<int> windowTimes(windows, 0);
    vector<vector<string>> windowTickets(windows);
    
    string cmd;
    while (true) {
        cout << "<<< ";
        cin >> cmd;
        
        if (cmd == "ENQUEUE") {
            int time;
            cin >> time;
            string ticket = generateTicket();
            visitors.push_back({time, ticket});
            cout << ">>> " << ticket << endl;
        }
        else if (cmd == "DISTRIBUTE") {
            // Вызываем функцию распределения
            distribute(visitors, windowTimes, windowTickets);
            
            // Выводим результаты
            for (int i = 0; i < windows; ++i) {
                if (!windowTickets[i].empty()) {
                    cout << ">>> Окно " << i+1 << " (" << windowTimes[i] << " минут): ";
                    for (size_t j = 0; j < windowTickets[i].size(); ++j) {
                        if (j != 0) cout << ", ";
                        cout << windowTickets[i][j];
                    }
                    cout << endl;
                }
            }
            break;
        }
        else {
            cout << ">>> Неизвестная команда\n";
        }
    }
    
    return 0;
}